compiler/rustc_codegen_ssa/src/back/symbol_export.rs

```diff
@@ -362,16 +400,30 @@ fn exported_symbols_provider_local(
                     }
                 }
                 MonoItem::Fn(Instance { def: InstanceKind::DropGlue(_, Some(ty)), args }) => {
-                    // A little sanity-check
-                    assert_eq!(args.non_erasable_generics().next(), Some(GenericArgKind::Type(ty)));
-                    symbols.push((
-                        ExportedSymbol::DropGlue(ty),
-                        SymbolExportInfo {
-                            level: SymbolExportLevel::Rust,
-                            kind: SymbolExportKind::Text,
-                            used: false,
-                        },
-                    ));
+                    let Some(GenericArgKind::Type(typ)) = args.non_erasable_generics().next()
+                    else {
+                        bug!("Expected a type argument for drop glue");
+                    };
```

Comment:
Why does this no longer check that `args.non_erasable_generics().next()` is equal to `ty`?

Reply:
That's an omission on my part. Will fix.